lib/std/heap.zig

```diff
@@ -479,6 +479,30 @@ pub fn StackFallbackAllocator(comptime size: usize) type {
     };
 }
 
+pub const DefaultAllocator = struct {
+    var debug_allocator: DebugAllocator(.{}) = .init;
+
+    pub const allocator = impl.gpa;
+    const impl: struct {
+        gpa: std.mem.Allocator,
+        debug: bool,
+    } = if (builtin.os.tag == .wasi)
+        .{ .gpa = wasm_allocator, .debug = false }
+    else if (builtin.link_libc)
+        .{ .gpa = c_allocator, .debug = false }
```

Comment:
There was some discussion on this over on Zulip: [#std > runtime page size and GPA enhancements @ 💬](https://zsf.zulipchat.com/#narrow/channel/454371-std/topic/runtime.20page.20size.20and.20GPA.20enhancements/near/497538699)

This point from Andrew is particularly relevant:

> oh yeah and probably DefaultAllocator should use libc_allocator for non-debug mode when linking libc

It implies that we still want `debug_allocator` in the `Debug` case when linking libc.